sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosRequestOptions.java

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosDiagnosticsThresholds;
+import com.azure.cosmos.CosmosEndToEndOperationLatencyPolicyConfig;
+import com.azure.cosmos.implementation.apachecommons.collections.list.UnmodifiableList;
+
+import java.util.List;
+
+/**
+ * The common request options for operations. This class should be used with the addPolicy method in the {@link com.azure.cosmos.CosmosClientBuilder}
+ *  to change request options without restarting the application.
+ *
+ */
+public final class CosmosRequestOptions implements ReadOnlyRequestOptions {
+    private CosmosEndToEndOperationLatencyPolicyConfig endToEndOperationLatencyConfig;
+    private ConsistencyLevel consistencyLevel;
+    private Boolean contentResponseOnWriteEnabled;
+    private Boolean nonIdempotentWriteRetriesEnabled;
+    private DedicatedGatewayRequestOptions dedicatedGatewayRequestOptions;
+    private String throughputControlGroupName;
+    private CosmosDiagnosticsThresholds thresholds;
+    private Boolean scanInQueryEnabled;
+    private List<String> excludeRegions;
+    private Integer maxDegreeOfParallelism;
+    private Integer maxBufferedItemCount;
+    private Integer responseContinuationTokenLimitInKb;
+    private Integer maxItemCount;
+    private Boolean queryMetricsEnabled;
+    private Boolean indexMetricsEnabled;
+    private Integer maxPrefetchPageCount;
+    private String queryName;
+
+    /**
+     * Sets the CosmosEndToEndLatencyPolicyConfig.
+     *
+     * @param endToEndOperationLatencyPolicyConfig the CosmosEndToEndLatencyPolicyConfig.
+     * @return current CosmosCommonRequestOptions.
```

Comment:
nit: staled class name. Also worth updating the samples in the description

Reply:
changed